src/plasma/plasma_manager.cc

```diff
@@ -595,13 +599,14 @@ void send_queued_request(event_loop *loop,
     break;
   case MessageType_PlasmaDataReply:
     LOG_DEBUG("Transferring object to manager");
-    if (conn->cursor == 0) {
-      /* If the cursor is zero, we haven't sent any requests for this object
+    if (ClientConnection_request_finished(conn)) {
+      /* If the cursor is not set, we haven't sent any requests for this object
        * yet, so send the initial data request. */
       err = handle_sigpipe(
           plasma::SendDataReply(conn->fd, buf->object_id.to_plasma_id(),
                                 buf->data_size, buf->metadata_size),
           conn->fd);
+      conn->cursor = 0;
```

Comment:
it seems that by providing `ClientConnection_request_finished` and `ClientConnection_finish_request`, you're trying to hide `conn->cursor` as an implementation detail. But that seems inconsistent with directly accessing `conn->cursor` here and in other places in this file.

Reply:
Ah it was more to make sure that we could keep the values consistent, in case we have to change it again and don't want to change the -1 check everywhere.